petab/v2/lint.py

```diff
@@ -621,11 +621,8 @@ def append_overrides(overrides):
         if not problem.model.has_entity_with_id(p)
     )
 
-    # remove parameters that occur in the condition table and are overridden
-    #  for ALL conditions
-    for p in problem.condition_df.columns[
-        ~problem.condition_df.isnull().any()
-    ]:
+    # parameters that are overridden via the condition table are not allowed
+    for p in problem.condition_df.columns:
```

Comment:
Since parameter IDs is now a set, can change try-except/loop to
```python
parameter_ids -= set(condition_df.columns)
```

Reply:
Since it's right before returning, it could be done just on the keys, but then future modifications might easily mess up the ordering. I'd leave it as is until Python has a proper OrderedSet.

Reply:
Where does it become `OrderedDict`? In the v1 code it's initialized as an `OrderedDict`, but here as a `set`?

Reply:
You are right. I was in v1.